app/main.py

```diff
@@ -1,2 +1,56 @@
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.capacity = 8   # default value
+        self.hash_table = self.create_hash_table()
+        self.load_factor = self.calculate_load_factor()
+        self.size = 0
+
+    def create_hash_table(self) -> list:
+        return [None] * self.capacity
+
+    def calculate_load_factor(self) -> int:
+        return int(self.capacity * (2 / 3))
+
+    def __setitem__(self, key: int, value: int) -> int:
+        hash_key = hash(key)
+        index = hash_key % self.capacity
+
+        while True:
+            if self.hash_table[index] is None:
+                self.hash_table[index] = (key, value, hash_key)
+                self.size += 1
+                if self.size == self.load_factor:
+                    self.resise_table()
+                break
+            elif self.hash_table[index][0] == key:
+                self.hash_table[index] = (key, value, hash_key)
+                break
+            index += 1
+            index %= self.capacity
```

Comment:
You can make while loop loke here : `while self.hash_table[index] is not None:` and this way your code will bw less nested